Arrow.swift

```diff
@@ -83,34 +83,47 @@ public func <-- <T>(inout left: [T]?, right: AnyObject?) {
 // MARK: - Parse Custom Types
 
 public protocol ArrowParsable {
-    init(json: JSON)
+    init()
+    mutating func deserialize(json:JSON)
 }
 
 infix operator <== {}
 public func <== <T:ArrowParsable>(inout left:T, right: AnyObject?) {
     if let r: AnyObject = right {
-        left = T.self(json:r)
+        var t = T.self()
```

Comment:
Maybe change these to `T.init()`? Same things but makes the intent more clear I think.